gateway/src/node/storage/mod.rs

```diff
@@ -222,7 +263,8 @@ impl PersistentStorage {
         Ok(PersistentStorage {
             shared_key_manager: SharedKeysManager::new(connection_pool.clone()),
             inbox_manager: InboxManager::new(connection_pool.clone(), message_retrieval_limit),
-            bandwidth_manager: BandwidthManager::new(connection_pool),
+            bandwidth_manager: BandwidthManager::new(connection_pool.clone()),
```

Comment:
oh. i see what's happening here. we have the same underlying database and just cloning the connection pool to different structs. I guess that could work. though not sure about scaling long-term, but I guess we'll worry about it when we cross that bridge

Reply:
I didn't think about it, I just blended in existing code